src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -54,17 +48,27 @@ public PersonCard(Person person, int displayedIndex) {
         this.person = person;
         id.setText(displayedIndex + ". ");
         name.setText(person.getName().fullName);
-        phone.setText(person.getPhone().value);
+
         if (person.hasAddress()) {
             Label address = new Label(person.getAddress().map(Object::toString).orElse(null));
             address.getStyleClass().add("cell_small_label");
             address.setText(person.getAddress().orElse(null).value);
             vBox.getChildren().add(address);
         }
 
-        email = new Label(person.getEmail().value);
-        email.getStyleClass().add("cell_small_label");
-        vBox.getChildren().add(email);
+        if (person.hasPhone()) {
+            Label phone = new Label(person.getPhone().map(Object::toString).orElse(null));
```

Comment:
the order of elements is slightly different from `Messages.java`, but it's not a bug so i think we can standardize this in another PR